src/components/TextField/TextField.test.tsx

```diff
@@ -227,6 +227,24 @@ test('it renders an inverse color icon when icon and tooltipMessage are provided
   expect(icon).toHaveClass('ChromaTextField-labelIconInverse');
 });
 
+test('it renders label when provided', async () => {
+  const { findByLabelText } = renderWithTheme(
+    <TextField label="label-text" data-testid={testId} />
+  );
+
+  const label = await findByLabelText(/label-text/);
+  expect(label).toBeInTheDocument();
+});
+
+test('it does not render label when not provided', async () => {
+  const { queryByLabelText } = renderWithTheme(
+    <TextField label="" data-testid={testId} />
+  );
+
+  const label = await queryByLabelText(/label-text/);
+  expect(label).not.toBeInTheDocument();
```

Comment:
Something else that may work, if you want to ensure no `label` elements at all is:

```suggestion
renderWithTheme(
  <TextField label="" data-testid={testId} />
);

expect(document.querySelectorAll('label').length).toEqual(0);
```

Reply:
Oh sweet! I will update this.